### ISSUE TITLE
Crash when Git can't auto-detect email

### ISSUE THREAD

**Issue by TheLastProject** [MycroftAI/mycroft-skills-kit#31]
```
$ msk create
[...]
Does this Skill depend on Python Packages (PyPI), System Packages (apt-get/others), or other skills?
This will create a manifest.yml file for you to define the dependencies for your Skill.
Check the Mycroft documentation at mycroft.ai/to/skill-dependencies to learn more about including dependencies, and the manifest.yml file, in Skills. (y/N) y
Traceback (most recent call last):
  File "/home/pi/mycroft-core/.venv/bin/msk", line 11, in <module>
    sys.exit(main())
  File "/home/pi/mycroft-core/.venv/lib/python3.7/site-packages/msk/__main__.py", line 67, in main
    return action_to_cls[args.action](args).perform()
  File "/home/pi/mycroft-core/.venv/lib/python3.7/site-packages/msk/actions/create.py", line 402, in perform
    self.commit_changes()
  File "/home/pi/mycroft-core/.venv/lib/python3.7/site-packages/msk/actions/create.py", line 349, in commit_changes
    self.git.commit(message='Initial commit')
  File "/home/pi/mycroft-core/.venv/lib/python3.7/site-packages/git/cmd.py", line 545, in <lambda>
    return lambda *args, **kwargs: self._call_process(name, *args, **kwargs)
  File "/home/pi/mycroft-core/.venv/lib/python3.7/site-packages/git/cmd.py", line 1014, in _call_process
    return self.execute(call, **exec_kwargs)
  File "/home/pi/mycroft-core/.venv/lib/python3.7/site-packages/git/cmd.py", line 825, in execute
    raise GitCommandError(command, status, stderr_value, stdout_value)
git.exc.GitCommandError: Cmd('git') failed due to: exit code(128)
  cmdline: git commit --message=Initial commit
  stderr: '
*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'pi@picroft.(none)')'
```

**Comment by krisgesling** [comment]
Thanks for flagging this Sylvia.

It also looks like the authentication mechanism MSK uses for Github is being deprecated so good timing for us to look into both.

**Comment by andlo** [comment]
I would look into this as when testing using github token it still is a problem if git cant determinate the username and useremail.....

So somewhere we need to check if git user.email and git user.name is set and if not ask for it and set it.
Alternative catch the error and error out with the same clear message without the traceback....

Just dont know where to start and where to look.....

**Comment by forslund** [comment]
This should be resolved now in #35. Closing.
